From e4577fd1d0011e5240f09b709b513492141c3585 Mon Sep 17 00:00:00 2001
From: Sebastian Vollnhals <sebastian@vollnhals.info>
Date: Fri, 27 Apr 2012 03:57:17 +0200
Subject: [PATCH] dropdown patch

---
 checkout_shipping.php |   50 ++++++++++++++++++++++++++++++++++++++++++++++++-
 1 files changed, 49 insertions(+), 1 deletions(-)

diff --git a/checkout_shipping.php b/checkout_shipping.php
index bc4acbe..3c23aaf 100755
--- a/checkout_shipping.php
+++ b/checkout_shipping.php
@@ -272,9 +272,54 @@ function rowOutEffect(object) {
 
 <?php
         } else {
+
+			if ($quotes[$i]['id'] == "tiramizoo") {
+				
+				echo '<tr class="moduleRow" onmouseover="rowOverEffect(this)" onmouseout="rowOutEffect(this)" onclick="selectRowEffect(this, 1); $(\'.tiramizoo-radio\').hide(); $(\'#tiramizoo-radio-\'+$(\'#tiramizoo-select\').val()+\'.tiramizoo-radio\').attr(\'checked\',\'checked\').show(); $(\'.tiramizoo-price\').hide(); $(\'#tiramizoo-price-\'+$(\'#tiramizoo-select\').val()+\'.tiramizoo-price\').show();"><td>';
+				
+				echo '<span style="float: left; line-height: 20px; height: 20px; margin: 0px 12px">'.MODULE_SHIPPING_TIRAMIZOO_TEXT_QUOTE_TITLE.'</span>';
+				
+				echo '<select size="1" name="tiramizoo-select" id="tiramizoo-select" onchange="$(\'.tiramizoo-radio\').hide(); $(\'#tiramizoo-radio-\'+this.value+\'.tiramizoo-radio\').attr(\'checked\',\'checked\').show(); $(\'.tiramizoo-price\').hide(); $(\'#tiramizoo-price-\'+this.value+\'.tiramizoo-price\').show();">';
+
+				$tiramizoo_radios = array();
+				$tiramizoo_prices = array();
+
+	         for ($j=0, $n2=sizeof($quotes[$i]['methods']); $j<$n2; $j++) {
+
+	            $checked = (($quotes[$i]['id'] . '_' . $quotes[$i]['methods'][$j]['id'] == $shipping['id']) ? true : false);				
+
+					echo '<option value="'.$quotes[$i]['id'] . '_' . $quotes[$i]['methods'][$j]['id'].'">'.strip_tags(array_pop(explode("<br /> ", $quotes[$i]['methods'][$j]['title']))).'</option>';
+
+					if ($j == 0) {
+						
+						$tiramizoo_prices[] = '<span class="tiramizoo-price" id="tiramizoo-price-'.$quotes[$i]['id'] . '_' . $quotes[$i]['methods'][$j]['id'].'">'.$currencies->format(tep_add_tax($quotes[$i]['methods'][$j]['cost'], (isset($quotes[$i]['tax']) ? $quotes[$i]['tax'] : 0))).'</span>';
+						$tiramizoo_radios[] = tep_draw_radio_field('shipping', $quotes[$i]['id'] . '_' . $quotes[$i]['methods'][$j]['id'], $checked, 'class="tiramizoo-radio" id="tiramizoo-radio-'.$quotes[$i]['id'] . '_' . $quotes[$i]['methods'][$j]['id'].'"');
+						
+					} else {
+						
+						$tiramizoo_prices[] = '<span style="display: none;" class="tiramizoo-price" id="tiramizoo-price-'.$quotes[$i]['id'] . '_' . $quotes[$i]['methods'][$j]['id'].'">'.$currencies->format(tep_add_tax($quotes[$i]['methods'][$j]['cost'], (isset($quotes[$i]['tax']) ? $quotes[$i]['tax'] : 0))).'</span>';
+						$tiramizoo_radios[] = tep_draw_radio_field('shipping', $quotes[$i]['id'] . '_' . $quotes[$i]['methods'][$j]['id'], $checked, 'class="tiramizoo-radio" style="display:none;" id="tiramizoo-radio-'.$quotes[$i]['id'] . '_' . $quotes[$i]['methods'][$j]['id'].'"');
+						
+					}
+
+				}
+
+				echo '</select>';
+								
+				echo '</td>';
+				
+				echo '<td>'.join("\n", $tiramizoo_prices).'</td>';
+				echo '<td align="right">'.join("\n", $tiramizoo_radios).'</td>';
+				
+				echo '</tr>';
+				
+				
+			} else {
+			
+
           for ($j=0, $n2=sizeof($quotes[$i]['methods']); $j<$n2; $j++) {
 // set the radio button to be checked if it is the method chosen
-            $checked = (($quotes[$i]['id'] . '_' . $quotes[$i]['methods'][$j]['id'] == $shipping['id']) ? true : false);
+            $checked = (($quotes[$i]['id'] . '_' . $quotes[$i]['methods'][$j]['id'] == $shipping['id']) ? true : false);				
 
             if ( ($checked == true) || ($n == 1 && $n2 == 1) ) {
               echo '      <tr id="defaultSelected" class="moduleRowSelected" onmouseover="rowOverEffect(this)" onmouseout="rowOutEffect(this)" onclick="selectRowEffect(this, ' . $radio_buttons . ')">' . "\n";
@@ -305,6 +350,9 @@ function rowOutEffect(object) {
       </tr>
 
 <?php
+
+	}
+
             $radio_buttons++;
           }
         }
-- 
1.7.8.3

